daemon/cluster/executor/container/container.go

```diff
@@ -207,7 +206,7 @@ func getMountMask(m *api.Mount) string {
 func (c *containerConfig) hostConfig() *enginecontainer.HostConfig {
 	return &enginecontainer.HostConfig{
 		Resources: c.resources(),
-		Binds:     c.bindMounts(),
+		Binds:     append(c.bindMounts(), c.volumes()...),
```

Comment:
The following should also work, given docker's current model:

`docker service create --mount type=volume,target=/foo busybox top`

It's going to be a big change to not automagically create volumes. These are declared this way all over the place, including in docker files.

The solution, as you have suggested, is to bind the lifecycle of these volumes to the scope of the container lifecycle.